test_terminating_delete.py

```diff
@@ -0,0 +1,312 @@
+                                         namespace=statefulset.metadata.namespace)
+        assert flag is True, "StatefulSet %s status check timed out, all replicas not ready..." % pvc_obj.metadata.name
+
+
+        #Get pod object so that we can get node of pod to shut down kubelet on
+        pod_obj = manager.hpe_list_pod_objects(statefulset.metadata.namespace, **labels)
+        nodename = manager.get_current_node_of_pod(pod_obj)
+        logging.getLogger().info("Nodename of the statefulset pod is %s" % nodename)
+        output = manager.stop_kubelet(nodename)
+
+        logging.getLogger().info("Sleeping for 7 minutes - let node get declared not ready and pods recreate")
+        sleep(420)
+        logging.getLogger().info("Over from sleeping... Starting kubelet back on node %s" % nodename)
+        
+        output = manager.start_kubelet(nodename) 
+        fields['field'] = fields['field'] + nodename
+        logging.getLogger().info("Field selector with nodename: %s" % fields['field'])
+
+        # Assertion for no pods left in terminating state
+        # This can fail if other pods on same worker node are in Terminating state
+        pod_terminating = manager.hpe_list_pod_objects(statefulset.metadata.namespace, **fields)
+        count_pod_terminating = len(pod_terminating.items)
+        assert count_pod_terminating is 0, "There are pods that are in Terminating state on the old worker node"
+
+        # Check that the new pods are running (on another node)
+        flag, sts_obj = manager.check_status(timeout, statefulset.metadata.name, kind='StatefulSet', status='Ready',
+                                         namespace=statefulset.metadata.namespace)
+        assert flag is True, "StatefulSet %s status check timed out, all replicas not ready on new node..." % pvc_obj.metadata.name
+        
+        pod_obj = manager.hpe_list_pod_objects(statefulset.metadata.namespace, **labels)
+        current_nodename = manager.get_current_node_of_pod(pod_obj)
+        logging.getLogger().info("Nodename after simulating kubelet down is %s" % current_nodename)
+
+        assert current_nodename is not nodename, "The two worker nodes are the same, pod did not recreate on another node"
+
+        assert manager.delete_statefulset(statefulset.metadata.name, statefulset.metadata.namespace)
+
+        assert manager.delete_sc(sc.metadata.name) is True
+
+        assert manager.check_if_deleted(timeout, sc.metadata.name, "SC", sc.metadata.namespace) is True, "SC %s is not deleted yet " \
+                                                                                  % sc.metadata.name
+    except Exception as e:
+        logging.getLogger().error("Exception in test_terminating_delete_statefulset :: %s" % e)
+        raise e
+
+    finally:
+        cleanup(None,sc,None,None,statefulset,None)
+        if nodename is not None:
+            manager.start_kubelet(nodename)
+		
+
+def test_terminating_deployment_with_pv():
```

Comment:
We are following the code formatting standards for our other Python automation projects:

Autoflake
Black
Flake8

This helps make the code easier to understand and review. If necessary, please apply these formatting standards to the code